#!/usr/bin/env python
PACKAGE = "mercury_joy"
from dynamic_reconfigure.parameter_generator_catkin import *
gen = ParameterGenerator()

button_names = ", ".join([
    "a_button",
    "b_button",
    "x_button",
    "y_button",
    "lb_button",
    "rb_button",
    "back_button",
    "start_button",
    "home_button",
    "left_axes_button",
    "right_axes_button"
])

axes_names = ", ".join([
    "left_x_axes",
    "left_y_axes",
    "lt_axes",
    "right_x_axes",
    "right_y_axes",
    "rt_axes",
])

gen.add("button_names", str_t, 0, "Button Names", button_names)
gen.add("axes_threshold", double_t, 0, "Axes Threshold", 0.1, 0, 1)
gen.add("axes_names", str_t, 0, "Axes Names", axes_names)
gen.add("axes_to_button", str_t, 0, "Configuration of your axes you wanted to act like a button", "".join([
    "(6,less-than,-0.5,arrow_right)",
    "(6,greater-than,0.5,arrow_left)",
    "(7,less-than,-0.5,arrow_down)",
    "(7,greater-than,0.5,arrow_up)",
]))
gen.add("button_to_axes", str_t, 0, "Configuration of your buttons you wanted to act like an axes", "")
gen.add("map_axes", str_t, 0, "Map value of axes", "".join([
    "(5,-1.0,1.0,1.0,0.0)",
    "(2,-1.0,1.0,1.0,0.0)",
]))

exit(gen.generate(PACKAGE, "mercury_joy", "Joy"))